sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -321,3 +328,97 @@ def test_cv_generator_support():
 
     sfs = SequentialFeatureSelector(knc, n_features_to_select=5, cv=splits)
     sfs.fit(X, y)
+
+
+# Metadata Routing Tests
+# ======================
```

Comment:
Can't this estimator actually be tested with the common tests we have?

Reply:
I don't think it worked when I tried. If you have a suggestion how we can incorporate them, please do share.

Reply:
You will need to add:

```python
    {
        "metaestimator": SequentialFeatureSelector,
        "estimator_name": "estimator",
        "estimator": "classifier",
        "X": X,
        "y": y,
        "estimator_routing_methods": ["fit"],
    },
```

Right now, I'm getting an error because we don't handle `**params` in the score method.

Reply:
I might miss something but there is nothing stating that the metaestimator should root the data to `estimator`. We should need to define the `get_metadata_routing` isn't it?

Reply:
The routing takes place in the `cross_val_score` method. That defines it own version of get_metadata_routing since its a function and not a class. I don't think we need to define `get_metadata_routing` in the class itself as that would be redundant.

Reply:
That raises an issue though. If passed metadata are not correct and routing validation fails, the error message would show a very confusing message to the user since it's coming from the `cross_val_score`. It probably makes sense to define the `get_metadata_routing` and do the validation in this estimator as well, to avoid that confusion maybe?

Reply:
I looked at it and I think considering the error message it might make sense to do the routing in the class even if it seems redundant. The only other way would be to send the error message from the class to the cross val score function in such a case but that seems to be a bit inconvenient and introduces unnecessary complexity.